src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration/ConfigurationClient.cs

```diff
@@ -4,44 +4,48 @@
 
 using Azure.Base.Diagnostics;
 using Azure.Base.Http;
+using Azure.Base.Http.Pipeline;
 using System;
 using System.Net.Http;
 using System.Threading;
 using System.Threading.Tasks;
 
-// TODO (pri 1): Add all functionality from the spec: https://msazure.visualstudio.com/Azure%20AppConfig/Azure%20AppConfig%20Team/_git/AppConfigService?path=%2Fdocs%2Fprotocol&version=GBdev
-// TODO (pri 1): Support "List subset of keys" 
-// TODO (pri 1): Support "Time-Based Access" 
-// TODO (pri 1): Support "KeyValue Revisions"
-// TODO (pri 1): Support "Real-time Consistency"
-// TODO (pri 2): Add support for filters (fields, label, etc.)
-// TODO (pri 2): Make sure the whole object gets deserialized/serialized.
-// TODO (pri 3): Add retry policy with automatic throttling
 namespace Azure.ApplicationModel.Configuration
 {
     public partial class ConfigurationClient
     {
-        const string SdkName = "Azure.Configuration";
-        const string SdkVersion = "1.0.0";
+        const string ComponentName = "Azure.Configuration";
+        const string ComponentVersion = "1.0.0";
+
+        static readonly HttpPipelinePolicy s_defaultRetryPolicy = RetryPolicy.CreateFixed(3, TimeSpan.Zero,
+            500, // Internal Server Error 
+            504  // Gateway Timeout
+        );
 
         readonly Uri _baseUri;
         readonly string _credential;
         readonly byte[] _secret;
-        HttpPipeline.Options _options;
-        HttpPipeline Pipeline;
+        HttpPipeline _pipeline;
+
+        public static HttpPipelineOptions CreateDefaultPipelineOptions()
+        {
+            var options = new HttpPipelineOptions(HttpClientTransport.Shared);
+            options.LoggingPolicy = LoggingPolicy.Shared;
+            options.RetryPolicy = s_defaultRetryPolicy;
+            return options;
+        }
 
         public ConfigurationClient(string connectionString)
-            : this(connectionString, options: new HttpPipeline.Options())
+            : this(connectionString, CreateDefaultPipelineOptions())
         {
         }
 
-        public ConfigurationClient(string connectionString, HttpPipeline.Options options)
+        public ConfigurationClient(string connectionString, HttpPipelineOptions options)
         {
             if (connectionString == null) throw new ArgumentNullException(nameof(connectionString));
             if (options == null) throw new ArgumentNullException(nameof(options));
 
-            _options = options;
-            Pipeline = HttpPipeline.Create(_options, SdkName, SdkVersion);
+            _pipeline = options.Build(ComponentName, ComponentVersion);
```

Comment:
Reading this line is a bit strange but not sure if it is because of the `options` or `build`... building options creates a pipeline?
Maybe changing `build` to `CreatePipeline?` or `pipelineOptions.Build`?

Reply:
Yeah, I agree it's a bit strange. But I am not sure what to do about it. I considered the following two options and rejected them:
1. Call the options type PipelineBuilder. I actually implemented it as such originally but then I did not like the fact that the user would never call Build (the client ctor calls it) and also that the user would not instantiate the builder (the client's CreatePipelineOptions method does). These two things make the type very diffrente than all other builders in the framework, where the users news up a builder, calls methods on it, and then calls "build". 
2. I also considered having HttpPipeline.Create method that takes the option. But I this means that pipeline has a dependency on options. Options has lots of smarts/policy that we might not like in the future, and pipeline is very simple (policy free). The moment pipeline depends on options, it now acquires all the baggage of options complexity. And I would prefer to keep the pipeline policy-free/simple.

Having said that, I agree with you on the issue and maybe we should continue thinking how to fix it.

Reply:
I do like this proposal where HttpPipeline.Create doesn't depend on options, so I agree with #2.

Interesting how naming brings such discussions :). I will keep thinking about it. It is not a blocking issue.